cmds/ping/ping.go

```diff
@@ -69,47 +76,96 @@ func beatifulLatency(before time.Time) (latency string) {
 	return latency
 }
 
+func cksum(bs []byte) uint16 {
+	sum := uint32(0)
+
+	for k := 0; k < len(bs)/2; k++ {
+		sum += uint32(bs[k*2]) << 8
+		sum += uint32(bs[k*2+1])
+	}
+	if len(bs)%2 != 0 {
+		sum += uint32(bs[len(bs)-1]) << 8
+	}
+	sum = (sum >> 16) + (sum & 0xffff)
+	sum = (sum >> 16) + (sum & 0xffff)
+	if sum == 0xffff {
+		sum = 0
+	}
+
+	return ^uint16(sum)
+}
+
+func ping1(netname string, host string, i uint64) {
+	c, derr := net.Dial(netname, host)
+	if derr != nil {
+		log.Fatalf("net.Dial(%v %v) failed: %v", netname, host, derr)
+	}
+	defer c.Close()
+	waitFor := time.Duration(*wtf) * time.Millisecond
+	c.SetDeadline(time.Now().Add(waitFor))
+	msg := make([]byte, *packetSize)
+	msg[0] = ICMP_TYPE_ECHO_REQUEST
+	msg[1] = 0
+	binary.BigEndian.PutUint16(msg[6:], uint16(i))
+	binary.BigEndian.PutUint16(msg[4:], uint16(i>>16))
+	binary.BigEndian.PutUint16(msg[2:], cksum(msg))
+	if _, err := c.Write(msg[:]); err != nil {
+		log.Printf("Write failed: %v", err)
```

Comment:
return fmt.Errorf
don't decide to print here.

Reply:
Done